### PR TITLE
refactor(command): refactor messaging handler

### PR DESCRIPTION
**Pull request by hahattan**
- update external commandQueryHandler function signature
- remove duplicate error message

Signed-off-by: Chris Hung <chris@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by hahattan** [comment]
recheck

**Comment by lenny-intel** [comment]
Note that the snap build is currently failing on all PRs. Snap team is aware and says we can merge without it passing.

File: internal/core/command/controller/messaging/external.go [hunk 1]
```diff
@@ -65,19 +62,23 @@ func commandQueryHandler(responseTopic string, qos byte, retain bool, dic *di.Co
...
-		responseEnvelope, edgexErr := getCommandQueryResponseEnvelope(requestEnvelope, deviceName, dic)
-		if edgexErr != nil {
-			responseEnvelope = types.NewMessageEnvelopeWithError(requestEnvelope.RequestID, edgexErr.Error())
+		responseEnvelope, err := getCommandQueryResponseEnvelope(requestEnvelope, deviceName, dic)
+		if err != nil {
+			responseEnvelope = types.NewMessageEnvelopeWithError(requestEnvelope.RequestID, err.Error())
 		}
 
+		messageBusInfo := container.ConfigurationFrom(dic.Get).MessageQueue
+		responseTopic := messageBusInfo.External.Topics[QueryResponseTopic]
```
Comment (lenny-intel) [inline internal/core/command/controller/messaging/external.go:79]: Do we need to verify `QueryResponseTopic` exists in `Topics`

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by lenny-intel** [comment]
@hahattan , some failed tests.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**